### PR TITLE
Retry Mechanism for Kubernetes Loader

### PR DESCRIPTION
**Pull request by DaveBben**
# Description

Retry mechanism using exponential backoff when trying to contact the Darp pod. This will make sure that an immediate fail wouldn't occur because envoy wasn't ready to accept connections yet.

## Issue reference

https://github.com/dapr/dapr/issues/689

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ X] Code compiles correctly
* [ ] Created/updated tests
* [X ] Unit tests passing
* [X ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/components/kubernetes_loader.go [hunk 1]
```diff
@@ -29,7 +30,29 @@ func NewKubernetesComponents(configuration config.KubernetesConfig) *KubernetesC
 
 // LoadComponents returns components from a given control plane address
 func (k *KubernetesComponents) LoadComponents() ([]components_v1alpha1.Component, error) {
+	b := backoff.NewExponentialBackOff()
+	b.MaxElapsedTime = 1 *time.Minute
```
Comment (youngbupark) [inline pkg/components/kubernetes_loader.go]: I would set `1 * timeMinute` as a const.

**Comment by yaron2** [comment]
Hey @DaveBben,  it'd be great if you can address the comments and we get this merged :)

Thanks!

File: pkg/components/kubernetes_loader.go [hunk 2]
```diff
@@ -29,7 +30,29 @@ func NewKubernetesComponents(configuration config.KubernetesConfig) *KubernetesC
 
 // LoadComponents returns components from a given control plane address
 func (k *KubernetesComponents) LoadComponents() ([]components_v1alpha1.Component, error) {
```
[the remark this answers is not in the document]
Comment (DaveBben) [reply pkg/components/kubernetes_loader.go]: Added a simple unit test for LoadComponents and requestControlPlane

File: pkg/components/kubernetes_loader.go (hunk 1 above)
Comment (DaveBben) [reply pkg/components/kubernetes_loader.go]: Added

File: pkg/components/kubernetes_loader.go [hunk 3]
```diff
@@ -63,10 +65,11 @@ func requestControlPlane(url string) ([]byte, error) {
 	}
 	err := client.Do(req, resp)
 	if err != nil {
-		// Request failed, retry again
+		// Request failed, try again
+		fmt.Println("trying again");
```
Comment (yaron2) [inline pkg/components/kubernetes_loader.go]: Can you import `log "github.com/Sirupsen/logrus"` and replace this with `log.Info("retrying getting components")`?

**Comment by yaron2** [comment]
> Made recommended changes

Thanks, left one last comment.

File: pkg/components/kubernetes_loader.go (hunk 3 above)
Comment (DaveBben) [reply pkg/components/kubernetes_loader.go]: Whoops, didn't mean to keep that in there. All good now

**Comment by yaron2** [comment]
Thank you for this contribution!

**Pull request merged**